### DISCUSSION TITLE
When using maverickjs adapter, using `tick()` inside `flushSync()` doesn't work as expected

### DISCUSSION THREAD

**Discussion by morgs32** [maxnowack/signaldb#903] [Q&A] [answered]
I was hoping that if I did an insert into a collection like this:
```
import { tick } from 'maverickjs/signal'

...
const resources = useReactivity(() => collection.findMany({}).fetch()})
const [state, setState] = useState(0)

console.log(resources)

onClick={() => {
   flushSync(() => {
      collection.insert(resource)
      tick()
   })
  setState(state+1)
})
```

That the next time it logged `resources` it would include the new one...

Here is a sandbox, without the `flushSync` which maybe is irrelevant to the question??
https://codesandbox.io/p/sandbox/empty-dawn-q3qd4t

**Comment by maxnowack** [comment]
Reactive updates in SignalDB are always scheduled to the next tick with `setTimeout(() => …, 0)` to improve performance. So I think what you want to achieve wouldn't work. 
Maybe you could explain your use case a bit more. I'm pretty sure there is another solution 🙂

**Reply by morgs32** [reply]
A good example I think is from this video at 19:00 when Ryan needs to use something like flushSync to make sure a new post is added to his kanban before he scrolls to the bottom of the list: https://youtu.be/-0Qi0yMyLwQ?si=PDm5620Zywani-mJ&t=1132

Or another is with drag and drop (like in a kanban board). Without a full sync state update then you probably have to keep some sort of "pending changes" state like I ask about here: https://github.com/electric-sql/electric/discussions/1400#discussioncomment-9862909

Let me know if this makes sense! Thanks for your help!

**Reply by maxnowack** [reply]
Reacts `flushSync` doesn't do the trick there as is don't know anything about the signal stuff. However it's thinkable to implement some option that could be passed to the data modification methods like  `insert`, `updateOne`, etc.  that do some similar thing and schedule the update in the current tick instead of the next one.

The drag&drop issue you mentioned is also related to https://github.com/hello-pangea/dnd/issues/424
I also use the `@hello-pangea/dnd` library and this issue bugs me a lot 😐 The easiest way to get around it, is like it's done in the electric-sql linearlite example, but this adds unnecessary complexity to the code.

**Reply by morgs32** [reply]
I'm with you on an option sort of like remix apparently does in that video.

Out of curiosity, does scheduling the updates in SignalDB for the next tick provide a meaningful benefit over the "batched updates" that maverickjs will do by default?? Maybe it's a behavior we don't need when using the maverickjs adapter?

**Reply by maxnowack** [reply]
That's something I also think about at the moment 😀 I'll try to figure out what my initial intention was with this and see if I could get rid of it.

**Reply by maxnowack** [reply]
I added the delay in #575 as a performance improvement when there were many changes. The issue was that the indices were rebuilt after every single change. However, since the trade-offs are significant compared to this relatively edge case of massive sequential updates, I’m removing the delay in #908. This PR also includes a `.batch()` method to ensure performance during large-scale updates.

**Reply by morgs32** [reply]
👍 appreciate the investigation and thumbs up for your solution. I can totally see using the `.batch()` method too.

**Reply by maxnowack** [reply, marked as answer]
Fix released in `v0.19.0` 🎉
